src/DIPS.Xamarin.UI.iOS/SheetContentView.cs

```diff
@@ -0,0 +1,59 @@
+﻿using DIPS.Xamarin.UI.Internal.Xaml.Sheet;
+using DIPS.Xamarin.UI.iOS;
+using UIKit;
+using Xamarin.Forms;
+using Xamarin.Forms.Platform.iOS;
+
+[assembly: ExportRenderer(typeof(SheetView), typeof(SheetContentView))]
+
+namespace DIPS.Xamarin.UI.iOS
+{
+    internal class SheetContentView : VisualElementRenderer<ContentView>
+    {
+        private UIView m_uiView;
+
+        protected override void OnElementChanged(ElementChangedEventArgs<ContentView> e)
+        {
+            base.OnElementChanged(e);
+
+            m_uiView = GetControl();
+
+            if (e.OldElement is SheetView oldElement)
+            {
+                oldElement.StateChanged -= OnSheetStateChanged;
+            }
+
+            if (e.NewElement is SheetView newElement)
+            {
+                newElement.StateChanged += OnSheetStateChanged;
+            }
+        }
+
+        private void OnSheetStateChanged(object sender, SheetView.SheetState state)
+        {
+            if (sender is SheetView sheetView && !sheetView.m_sheetBehaviour.InterceptDragGesture)
+            {
+                return;
+            }
+            
+            var subviews = m_uiView.Subviews;
+            
+            ToggleScrollViews(subviews, state is SheetView.SheetState.Maximized);
+        }
+
+        private static void ToggleScrollViews(UIView[] subviews, bool scrollEnabled)
+        {
+            foreach (var view in subviews)
+            {
+                if (view is UIScrollView scrollView)
```

Comment:
Will this work if a subview has subviews? Like a grid with scrollview and a label.

Reply:
Yes, it's a recursive loop. It will go through all subviews of subviews until it finds a scrollview. Only assumes one scrollview/or that the outer scroll view is the vertical one. If there is a horizontal scrollview with a vertical scrollview only the horizontal scrollview will be disabled. We could support that as well.